lib/chef/provider/package/chocolatey.rb

```diff
@@ -220,7 +233,10 @@ def parse_list_output(*args)
           hash = {}
           choco_command(*args).stdout.each_line do |line|
             name, version = line.split('|')
-            hash[name.downcase] = version.chomp
+            version = version.chomp
+            if desired_name_versions[name].nil? || desired_name_versions[name] == version
+              hash[name.downcase] = version
+            end
```

Comment:
so the `list -ar` lists all available versions, then we're assured that if we want to install version 1.2.3 that it'll eventually match here, even though later versions may be available?

what about for installed packages?  i think it may superficially get the behavior right since if you've got a version that doesn't match it'll return nil like its uninstalled, but that does get the semantics of the current_resource.version wrong.